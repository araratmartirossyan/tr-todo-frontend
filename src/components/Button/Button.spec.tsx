import { shallow, ShallowWrapper } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Button } from './index'

describe('Button test', () => {
  let component: ShallowWrapper
  let mockCallback: jest.Mock

  beforeEach(() => {
    mockCallback = jest.fn()
    component = shallow(<Button onClick={mockCallback}>Some text</Button>)
  })

  it('should render Button component', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render props text', () => {
    expect(component.text()).toBe('Some text')
  })

  it('should  call event on click', () => {
    const btn = component.find('[data-test-id="todo-button"]')
    expect(btn.exists()).toBe(true)
    btn.simulate('click')
    expect(mockCallback.mock.calls.length).toBe(1)
  })
})
