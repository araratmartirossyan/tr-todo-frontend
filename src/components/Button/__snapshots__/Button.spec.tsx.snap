// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test should render Button component 1`] = `
<styled.button
  data-test-id="todo-button"
  onClick={[MockFunction]}
>
  Some text
</styled.button>
`;
